src/context.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+
+interface PortalProviderProps {
+  children?: React.ReactNode;
+}
+
+export interface PortalValue {
+  setPortal: (name: string, children: React.ReactNode) => void;
+  getPortal: (name: string) => React.ReactNode | undefined;
+}
+
+export const PortalContext = React.createContext<PortalValue | null>(null);
+
+export function PortalProvider(props: PortalProviderProps): JSX.Element {
+  const [portals, setPortals] = React.useState<Map<string, React.ReactNode>>(
+    new Map()
```

Comment:
Is there a good reason to prefer `Map` here? I wonder if this would exclude React Native and some other fringe environments for not a lot of gain over using a regular object?

Reply:
I'm cool with using a regular object but `Map` has near universal support.

Reply:
I can change it though.

Reply:
6ac67d5

Reply:
Even in React Native (which uses an embedded JavaScript interpreter, I think)?